src/components/body/EmojiList.tsx

```diff
@@ -3,9 +3,13 @@ import categories from '../../dataUtils/categories';
 import { EmojiCategory } from './EmojiCategory';
 import './EmojiList.css';
 
-export function EmojiList() {
+interface Props { 
+  emojiListRef: any;
+}
+
+export function EmojiList({ emojiListRef } : Props) {
   return (
-    <ul className="epr-emoji-list">
+    <ul className="epr-emoji-list" ref={emojiListRef}>
```

Comment:
Now that I look further down the PR, I see that we're drilling our refs over and over again. Wouldn't it be simpler to create a new context only containing element references?

You can create a new `<ElementRefConfetProvider>` component that handles propagating these references down, and then add hooks that tap into the context to just grab these elements from the context.

**Why?**
While drilling is harmless, it adds one logistical issue. It makes components that should be unaware of these refs - aware of these refs.

`Body` does not _need_ to know about emojiListRef, just as `Header` doesn't need to know about the search and skin tones to perform its function.